agent/proxycfg/testing_ingress_gateway.go

```diff
@@ -1485,6 +1485,164 @@ func TestConfigSnapshotIngressGateway_SingleTLSListener(t testing.T) *ConfigSnap
 		})
 }
 
```

Comment:
I have a question, is it possible to have an Ingress gateway listener that listens to port x for grpc and port y for http2?

Reply:
A listener only binds to a single port. It is possible to configure an ingress gateway with one listener on port X with the protocol as `grpc`, and different listener on port Y with the protocol configured as `http2`.

Reply:
Thanks @blake. @oulman could you add a test function for this use case?

Reply:
Sure! I just added a test for `TestConfigSnapshotIngressGateway_MultiTLSListener_MixedHTTP2gRPC` which validates two listeners (grpc and http2) with per-listener TLSConfig's.

The other scenario this made me think about is configuring TLS at the gateway [level](https://www.consul.io/docs/connect/config-entries/ingress-gateway#tls). From my read it looks like the Gateway level TLS is still merged down to the listener level and setting ALPN in `makeDownstreamTLSContextFromSnapshotListenerConfig()` still applies. I added another test `TestConfigSnapshotIngressGateway_GWTLSListener_MixedHTTP2gRPC` to validate that case. 

Thanks!